test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -1342,9 +1351,30 @@ func getClusterwideStatus(c clientset.Interface) (string, error) {
 }
 
 type scaleUpStatus struct {
-	status string
-	ready  int
-	target int
+	status    string
+	ready     int
+	target    int
+	timestamp time.Time
+}
+
+// Try to get timestamp from status.
+// Status configmap is not parsing-friendly, so evil regexpery follows.
+func getStatusTimestamp(status string) (time.Time, error) {
+	timestampMatcher, err := regexp.Compile("Cluster-autoscaler status at \\s*([0-9\\-]+ [0-9]+:[0-9]+:[0-9]+\\.[0-9]+ \\+[0-9]+ [A-Za-z]+):")
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	timestampMatches := timestampMatcher.FindAllStringSubmatch(status, -1)
```

Comment:
Nit: Could this be replaced with `timestampMatcher.FindStringSubmatch`? I think it would save you the writing `timestampMatches[0]` in the `timestamp, err := time.Parse(timestampFormat, timestampMatches[0][1]) line.

The `if len(timestampMatches) < 1` could be replaced with `if timestampMatch == nil`.

Entirely up to you though!

Reply:
Good point, thanks:) This was already approved and merged before I could change this, so I'll probably submit another PR

Reply:
Sweet :) I'm happy so submit a pr as well if you'd like @aleksandra-malinowska - I think it could be a great way for me to get more familiar with the autoscaling e2e tests. But also, no worries if you already have plans to do it!

Reply:
PR here - https://github.com/kubernetes/kubernetes/pull/52944